sklearn/linear_model/cd_fast.pyx

```diff
@@ -270,89 +392,198 @@ def enet_coordinate_descent(np.ndarray[floating, ndim=1] w,
                 if fabs(w[ii]) > w_max:
                     w_max = fabs(w[ii])
 
+            # Variable screening
             if (w_max == 0.0 or
                 d_w_max / w_max < d_w_tol or
-                n_iter == max_iter - 1):
+                n_iter == max_iter - 1 or
+                screening > 0 and n_iter % screening == 0) and n_iter != 0:
                 # the biggest coordinate update of this iteration was smaller
                 # than the tolerance: check the duality gap as ultimate
                 # stopping criterion
 
-                # XtA = np.dot(X.T, R) - beta * w
-                for i in range(n_features):
-                    XtA[i] = dot(n_samples, &X_data[i * n_samples],
-                                 1, R_data, 1) - beta * w[i]
+                gap = enet_duality_gap(n_samples, n_features, n_tasks,
+                                       X_data, y_data, R_data, w_data, XtA_data,
+                                       dual_scaling_data, alpha, beta, positive,
+                                       <np.int32_t*>disabled.data,
```

Comment:
Is the cast necessary there ? It seems inconsistent with `sparse_enet_duality_gap`